### PR TITLE
Video support improvements

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | tba
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Add two new improvements to video integration:
- add images for .mp4  player
- remove dependencies on form-id - now we can use video with tracking without form gated functionality

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1.  Create gated video for mp4 

```
<video width="640" height="360" data-form-id="1" data-gate-time="15">
    <source type="video/mp4" src="https://server.tld/video.mp4" />
</video>
```
2.  Player without controls. See browser console for no 404 calls

#### Steps to test this PR for bug:
1. Apply PR and repeat steps to reproduce the bug:
2. See video player with controls

#### Steps to test this PR for new feature (remove dependencies on form-id):
1. Apply PR  and use video without data-form-id param
```
<video width="640" height="360"  data-gate-time="0">
    <source type="video/mp4" src="https://server.tld/video.mp4" />
</video>
```
2. Play video and see video play hit on contact timeline
3. Check browser console for no 404 calls

### CODE REVIEW

File: app/bundles/PageBundle/EventListener/BuildJsSubscriber.php [hunk 1]
```diff
@@ -382,12 +378,15 @@ public function onBuildJsForVideo(BuildJsEvent $event)
         }
        
         var cookieName = 'mautic-player-'+i+'-'+node.dataset.formId;
```
Comment (dongilbert) [inline app/bundles/PageBundle/EventListener/BuildJsSubscriber.php]: I think we need to generate a random cookie name here for players without an associated form - what are your thoughts?

Comment (kuzmany) [inline app/bundles/PageBundle/EventListener/BuildJsSubscriber.php]: I moved this away we could use it just with forms

**Comment by mqueme** [comment]
This worked for me.

**Comment by dongilbert** [comment]
Works well! Thanks @kuzmany

**Pull request merged**